File: python/2.7/factory/PizzaStore.py
from abc import ABCMeta, abstractmethod

class PizzaStore:
    __metaclass__ = ABCMeta

    @abstractmethod
    def createPizza(item):
        pass

    def orderPizza(self, type):
        self.pizza = self.createPizza(type)
        print "--- Making a %s ---" %(self.pizza.getName())
        self.pizza.prepare()
        self.pizza.bake()
        self.pizza.cut()
        self.pizza.box()
        return self.pizza

File: php/5.3/factory/com/pizza/pizzaStore/PizzaStore.php
<?
abstract class PizzaStore {
    abstract public function createPizza($item);

    public function orderPizza($type) {
        $pizza = $this->createPizza($type);
        echo ("--- Making a " . $pizza->getName() . " ---\n");
        $pizza->prepare();
        $pizza->bake();
        $pizza->cut();
        $pizza->box();
        return $pizza;
    }
}
?>